File: Problemas/1542A - Odd Set/1542A_Odd_Set.cpp
// XXXA - 
// https://codeforces.com/problemset/problem/XXX/A

#include <stdio.h>
#include <stdlib.h>
#include <bits/stdc++.h>
#include <cmath>
using namespace std;

int main(){

    // Input:
    // 1) pruebas = Número de casos de prueba
    int pruebas;
    cin >> pruebas;

    // Itera sobre cada caso de prueba
    for(int indice_prueba = 0; indice_prueba < pruebas; indice_prueba++) {

        // Input:
        // 2) N = Número de elementos
        int N;
        cin >> N;

        // Input:
        // 2) multiset = Números a emparejar
        int multiset[201];
        for(int indice_numero = 0; indice_numero < 2*N; indice_numero++) {
            cin >> multiset[indice_numero];
        }

        // Contador de números impares
        int impares = 0;

        // Itera sobre los números del multiset
        for(int indice_numero = 0; indice_numero < 2*N; indice_numero++) {
            
            //Si el número es impar, entonces suma +1 al contador de impares
            if(multiset[indice_numero] % 2 == 1) {
                impares++;
            }
        }

        // Si el contador de números impares es igual a N, entonces se pueden generar parejas de sumas impares
        // Imprime "YES" si se pueden formar parejas de sumas impares, de lo contrario, imprime "NO"
        if(impares == N) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }
    }

    return 0;
}

File: Problemas/1542A - Odd Set/1542A_Odd_Set.py
# 1542A - Odd Set
# https://codeforces.com/problemset/problem/1542/A

# Input:
# 1) pruebas = Número de casos de prueba
pruebas = int(input())

# Itera sobre cada caso de prueba
for indice_prueba in range(0,pruebas):

    # Input:
    # 2) N = Número de elementos
    N = int(input())

    # Input:
    # 3) multiset = Números a emparejar
    multiset = list(map(int, input().split()))

    # Contador de números impares
    impares = 0

    # Itera sobre los números del multiset
    for numero in multiset:

        # Si el número es impar, entonces suma +1 al contador de impares
        if numero % 2 == 1:
            impares += 1

    # Si el contador de números impares es igual a N, entonces se pueden generar parejas de sumas impares
    # Imprime "YES" si se pueden formar parejas de sumas impares, de lo contrario, imprime "NO"
    if impares == N:
        print("YES")
    else:
        print("NO")

